src/common.h

```diff
@@ -8,6 +8,12 @@
 #define __DCURL_MINOR__ 1
 #define __DCURL_PATCH__ 0
 
+#if defined(ENABLE_DEBUG)
+#define DPRINTF(...) printf(__VA_ARGS__)
```

Comment:
Lower-case `dprintf` is better than the opposite form `DPRINTF`.

Reply:
> Eventually, `dprintf` might be conjuncted with logging facilities such as something in `entangle`. I would expect there is no direct call to `printf`.

I guess it will be left to the future modification?

Reply:
We can proceed at the moment since it could be involved in small chunks of changes.

Reply:
I found the logging function of `entangled`: [log_debug()](https://github.com/iotaledger/entangled/blob/09f58f17072053798b0d4adec0b1263eb005641c/utils/logger_helper.h#L23)

```c
static inline void dprintf(...) {
#if defined(ENABLE_DEBUG)
    log_debug();
#endif
}

void log_debug() {
    // The current dcurl implementation would be printf()
    // In the future,
    // it might be integrated with other logging facilities like entangled
    printf(...);
}
```

Is this what you expected to see?

Reply:
`selftest` programs inside Linux kernel tree come with a nice example for writing dprintf: https://github.com/torvalds/linux/blob/master/tools/testing/selftests/x86/pkey-helpers.h

Reply:
Since `dprintf()` has been declared in **stdio.h**, I rename it to `dbgprintf()` instead.